### DISCUSSION TITLE
How to add values of several textboxes

### DISCUSSION THREAD

**Discussion by hulcon** [alpinejs/alpine#1361] [1. Help]
Hi! I am using the TALL stack (including Livewire and Alpine JS ) for the first time in one of my projects.
![trash](https://user-images.githubusercontent.com/28725843/120436920-28fec880-c39d-11eb-9028-c6f1692c95d8.jpg)

I need to add values of all boys text fields together (General boys, SC boys, ST boys) as well as that of girls text fields.
Although it is fairly easy to do that in Livewire but I don't want a server request to be fired for a simple arithmetic addition. I am unable to figure out how to achieve it in AlpineJs. The categories (General boys, SC boys and ST boys) are not hard-coded. They are in an array of the Livewire component. Here is the overview of the array:
![object](https://user-images.githubusercontent.com/28725843/120438705-551b4900-c39f-11eb-9115-b680009d612e.jpg)

and here is the Livewire component view:
`
<div class="flex flex-col w-full lg:grid lg:grid-cols-2 xl:grid-cols-3">
            @forelse ($classwiseEnrolmentDetails['enrolmentDetails'] as $classDetail)
                <div class="bg-gradient-to-r from-gray-800 to-gray-700 rounded-md shadow-md mx-3 my-3 flex flex-col" >
                    <div class="my-2">
                        <div class="flex flex-col my-2 mx-3">
                            <div class="flex flex-row w-full items-center justify-center text-center text-2xl font-bold text-white mb-1">
                                Class {{ $classDetail['classLabel'] }}
                            </div>
                            <div class="flex flex-row w-full items-center px-3 py-2 mb-1 bg-gray-800 bg-opacity-80 rounded-md">
                                <div class="flex flex-col w-1/5 mx-1">
                                    <label class="flex font-bold text-green-400">Category</label>
                                </div>
                                <div class="flex flex-col w-2/5 items-center mx-1">
                                    <label class="flex font-bold text-indigo-400">Boys</label>
                                </div>
                                <div class="flex flex-col w-2/5 items-center mx-1">
                                    <label class="flex font-bold text-pink-400">Girls</label>
                                </div>
                            </div>
    
                            @forelse ($classDetail['enrolment'] as $enrolment)
                                <div class="flex flex-row items-center w-full px-3 py-2 mb-1 bg-indigo-700 bg-opacity-30 rounded-md" >
    
                                    <div class="flex flex-col w-1/5 mx-1">
                                        <label class="font-semibold text-green-100">{{ $enrolment['categoryName'] }}</label>
                                    </div>
                                    <div class="flex flex-col w-2/5  mx-1">
                                        <x-jet-input
                                            type="text"
                                            placeholder="{{ $enrolment['categoryName'] }} Boys"
                                            wire:model.defer="{{ 'classwiseEnrolmentDetails.enrolmentDetails.' . $loop->parent->index . '.enrolment.' . $loop->index . '.boys' }}"
    
                                            x-ref="{{ 'enrolment_' . $loop->parent->index . '_' . $loop->index . 'boys'}}"
                                            />
                                        <x-jet-input-error
                                            class="mt-1 font-bold"
                                            for="{{ 'classwiseEnrolmentDetails.enrolmentDetails.' . $loop->parent->index . '.enrolment.' . $loop->index . '.boys' }}" />
    
                                    </div>
                                    <div class="flex flex-col w-2/5 mx-1">
                                        <x-jet-input
                                            type="text"
                                            placeholder="{{ $enrolment['categoryName'] }} Girls"
                                            wire:model.defer="{{ 'classwiseEnrolmentDetails.enrolmentDetails.' . $loop->parent->index . '.enrolment.' . $loop->index . '.girls' }}"
										    
                                            x-ref="{{ 'enrolment_' . $loop->parent->index . '_' . $loop->index . 'girls'}}"
                                            />
                                        <x-jet-input-error
                                            class="mt-1 font-bold"
                                            for="{{ 'classwiseEnrolmentDetails.enrolmentDetails.' . $loop->parent->index . '.enrolment.' . $loop->index . '.girls' }}" />
                                    </div>
                                </div>
                            @empty
                                
                            @endforelse
    
                            <div class="flex flex-row items-center w-full my-2 px-3 py-1 bg-yellow-100 rounded-md border-2 border-yellow-300">
                                <div class="flex flex-col w-1/5 mx-1 font-bold">
                                    Total
                                </div>
                                <div class="flex flex-col w-2/5 mx-1 font-semibold">
                                    Boys: Boys Total needs to be displayed here
                                </div>
                                <div class="flex flex-col w-2/5 mx-1 font-semibold">
                                    Girls: Girls Total needs to be displayed here
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
            @empty
    </div>
`

Whenever a user provides input in any one of the text fields (girls or boys), the corresponding sum of the girls/boys should get updated. Although i do know that AlpineJs has @input but i couldn't figure out how i could add all the text fields on this event. The problem is that since the number of text fields are variable, i don't know how to loop over all the x-references on the @input event or at the div where i need to display the total. Kindly help

**Comment by KevinBatdorf** [comment]
Maybe a better question to ask on the Livewire repo. 

I personally don't know how wire:model works with `x-model`. You could collect all the relevant `x-ref` values and add them up. Or handle it on the server via Liveware and bind the result to an alpine value via `@entangle`?
